shared/src/store/action-reducers/simulated-region.ts

```diff
@@ -29,6 +31,95 @@ import { TransferPointActionReducers } from './transfer-point';
 import { isCompletelyLoaded } from './utils/completely-load-vehicle';
 import { getElement, getElementByPredicate } from './utils/get-element';
 
+/**
+ * This function assumes that every SimulatedRegion holds **ONLY** material,vehicles,patients and personnel
+ * (in any amount) and **ONE** TransferPoint.
+ * It only deletes material,vehicles and personnel if they are **ALL** in the SimulatedRegion to be deleted
+ * If one were to add more things to a SimulatedRegion one would have to change this function accordingly.
+ * @param draftState: The Draft State from which the SimulatedRegion should be deleted.
+ * @param simulatedRegionId: The Id of the SimulatedRegion that should be deleted.
+ */
+
+export function deleteSimulatedRegion(
+    draftState: Mutable<ExerciseState>,
+    simulatedRegionId: UUID
+) {
+    // Find related vehicles
+
+    const relatedVehicles = Object.values(draftState.vehicles).filter(
+        (vehicle) =>
+            isInSpecificSimulatedRegion(vehicle, simulatedRegionId) &&
+            Object.keys(vehicle.materialIds).every((materialId) =>
+                isInSpecificSimulatedRegion(
+                    getElement(draftState, 'material', materialId),
+                    simulatedRegionId
+                )
+            ) &&
+            Object.keys(vehicle.personnelIds).every((personnelId) =>
+                isInSpecificSimulatedRegion(
+                    getElement(draftState, 'personnel', personnelId),
+                    simulatedRegionId
+                )
+            )
+    );
+
+    // Find and delete related materials and personnel
+
+    Object.values(draftState.materials)
+        .filter(
+            (material) =>
+                isInSpecificSimulatedRegion(material, simulatedRegionId) &&
+                relatedVehicles.includes(
+                    getElement(draftState, 'vehicle', material.vehicleId)
+                )
+        )
+        .forEach((material) => delete draftState.materials[material.id]);
+
+    Object.values(draftState.personnel)
+        .filter(
+            (personnel) =>
+                isInSpecificSimulatedRegion(personnel, simulatedRegionId) &&
+                relatedVehicles.includes(
+                    getElement(draftState, 'vehicle', personnel.vehicleId)
+                )
+        )
+        .forEach((personnel) => delete draftState.personnel[personnel.id]);
```

Comment:
You have already checked that `relatedVehicles` only includes vehicles where all materials and personnel of that vehicle are in the current simulated region. So I think the easier way would be to so something like (pseudocode, might have to be fixed to fit this situation):

```ts
realtedVehicles.forEach((vehicle) => {
    vehicle.materialIds.forEach((id) => delete draftState.material[id]);
    vehicle.personnelIds.forEach((id) => delete draftState.personel[id]);
}
```

Reply:
Done